lib/ansible/config/manager.py

```diff
@@ -101,10 +102,14 @@ def ensure_type(value, value_type, origin=None, origin_ftype=None):
             value = boolean(value, strict=False)
 
         elif value_type in ('integer', 'int'):
-            value = int(value)
-
+            if not isinstance(value, int):
+                if (decimal_value := decimal.Decimal(value)) == (int_part := int(decimal_value)):
```

Comment:
This can fail with an `decimal.InvalidOperation` exception, but the caller (`get_config_value_and_origin`) is only expecting a `ValueError`.

Reply:
The same issue probably exists in the other locations this pattern is used.

Reply:
used the underlying decimal.DecimalException, now just need to add tests.

Thought bout just having all use the same function, but I'll leave that for another PR to start consolidating the 'type casting/validation' systems